openpype/hosts/nuke/plugins/publish/extract_thumbnail.py

```diff
@@ -179,10 +179,13 @@ def render_thumbnail(self, instance, output_name=None, **kwargs):
         temporary_nodes.append(write_node)
         tags = ["thumbnail", "publish_on_farm"]
 
+        thumb_output_name = "thumb"
+        if output_name:
+            thumb_output_name = "{}_{}".format(output_name, thumb_output_name)
         repre = {
             'name': name,
```

Comment:
No - this shouldn't be allowed. Representation names must be **unique** - if we're not updating name per thumbnail and it remains `thumbnail` then this is breaking the backend as far as I know.

E.g. [this function](https://github.com/pypeclub/OpenPype/blob/73e928efc99dbcd90b84d09cd1de0397ed3f633b/openpype/client/server/entities.py#L464-L475) always returns one entity, and if names are not unique it'd be ambiguous to know which one.

As far as I'm aware representation names are required to be unique - _but_ correct me if I'm wrong.

I'd also be quite confused seeing the Loader showing `action (thumbnail)` and `action (thumbnail)` twice not knowing which is which thumbnail. (Loader shows the Loader label + the representation name)

Reply:
Usually it's a single thumbnail for a version - and thus one representation. If the thumbnails are related to the "review" and we want to `link` it to that review maybe having the representations named after the thing its reviewing.

In practice however, I'm not entirely convinced we really _need_ a thumbnail per representation.

Anyway, I don't have a clear idea on how it should be done. Just remarking that by design **representation names must be unique** as far as I know and that this would make it so that _that_ is in conflict.

Reply:
BTW why is thumbnail integrated (as representation)? Is it actually used for loading?

Reply:
> BTW why is thumbnail integrated (as representation)? Is it actually used for loading?

We need multiple thumbnail representation only for Ftrack/Shotgun integration. Those are added as asset version components. After that we can probably keep only one. And we do not need to integrate them as representation to be shown in Loader - so if there is a way to remove them it would be great.

Reply:
> After that we can probably keep only one. **And we do not need to integrate them as representation to be shown in Loader** - so if there is a way to remove them it would be great.

You meant only the "duplicated" ones, right? Since we'd still need thumbnails as representation currently as that is how the Loader UI is designed to show it or am I wrong there? Plus that's how I suppose site sync syncs it?

Reply:
So in conclussion following had been added via this commit d2729775afdacfa73d703e7c15ba48d5a53715e7:
- added `delete` tag so it is not integrated into db as representation
- adding output preset name to thumb name only if multiple bake streams
- adding thumbnails to explicit cleanup paths - so they are removed after integration
- thumbnail file name inherited from representation name